src/main/java/de/tum/cit/aet/artemis/communication/web/FaqResource.java

```diff
@@ -0,0 +1,179 @@
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastInstructor;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+import de.tum.cit.aet.artemis.core.service.AuthorizationCheckService;
+import de.tum.cit.aet.artemis.core.util.HeaderUtil;
+
+/**
+ * REST controller for managing Faqs.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class FaqResource {
+
+    private static final Logger log = LoggerFactory.getLogger(FaqResource.class);
+
+    private static final String ENTITY_NAME = "faq";
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    private final CourseRepository courseRepository;
+
+    private final FaqRepository faqRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    public FaqResource(CourseRepository courseRepository, AuthorizationCheckService authCheckService, FaqRepository faqRepository) {
+
+        this.courseRepository = courseRepository;
+        this.authCheckService = authCheckService;
+        this.faqRepository = faqRepository;
+    }
+
+    /**
+     * POST /faqs : Create a new faq.
+     *
+     * @param faq the faq to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new faq, or with status 400 (Bad Request) if the faq has already an ID
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("faqs")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Faq> createFaq(@RequestBody Faq faq) throws URISyntaxException {
+        log.debug("REST request to save Faq : {}", faq);
+        if (faq.getId() != null) {
+            throw new BadRequestAlertException("A new faq cannot already have an ID", ENTITY_NAME, "idExists");
+        }
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, faq.getCourse(), null);
+
+        Faq savedFaq = faqRepository.save(faq);
+        return ResponseEntity.created(new URI("/api/faqs/" + savedFaq.getId())).body(savedFaq);
+    }
+
+    /**
+     * PUT /faqs/{faqId} : Updates an existing faq.
+     *
+     * @param faq   the faq to update
+     * @param faqId id of the faq to be updated
+     * @return the ResponseEntity with status 200 (OK) and with body the updated faq, or with status 400 (Bad Request) if the faq is not valid, or with status 500 (Internal
+     *         Server Error) if the faq couldn't be updated
+     */
+    @PutMapping("faqs/{faqId}")
+    @EnforceAtLeastInstructor
+    public ResponseEntity<Faq> updateFaq(@RequestBody Faq faq, @PathVariable Long faqId) {
+        log.debug("REST request to update Faq : {}", faq);
+        if (faqId == null || !faqId.equals(faq.getId())) {
+            throw new BadRequestAlertException("Id of FAQ and path must match", ENTITY_NAME, "idNull");
+        }
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, faq.getCourse(), null);
```

Comment:
Should we fully trust the caller to provide the correct course in the request body? I feel like this method doesn't guarantee that the Faq belongs to the course it claims to be from.

We should also consider using a DTO

Reply:
integrated a check fr the ID in each method, also added the corresponding test cases